enhancements/api-review/custom-tags-aws.md

```diff
@@ -0,0 +1,126 @@
+ - tags must be applied at creation time, in an atomic operation. It isn't acceptable to create an object and,
+   after some period of time, apply the tags post-creation.
+
+### Non-Goals
+
+ - to reduce initial scope, tags are applied only at creation time and not reconciled. If an administrator manually
+   changes the tags stored in the infrastructure resource, behavior is undefined. See below.
+ - to reduce initial scope, we are not implementing this for clouds other than AWS. We will not take any actions
+   to prohibit that later.
+
+## Proposal
+
+New `experimentalPropagateUserTags` field added to `.platform.aws` of install config to indicate that the user tags should be applied to AWS
+resources created by in-cluster operators.
+
+If `experimentalPropagateUserTags` is true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+
+Add a new field `userTags` to `.status.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
+`userTags` field will be applied to new resources created for the cluster. The `userTags` field will be populated by the installer only if the `experimentalPropagateUserTags` field is true.
+
+Installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances) (already exists)
+
+All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
+will apply these tags to all AWS resources they create.
+
+userTags that are specified in the infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
+
+The userTags field is intended to be set at install time and is considered immutable. Components that respect this field must only ever add tags that they retrieve from this field to cloud resources, they must never remove tags from the existing underlying cloud resource even if the tags are removed from this field(despite it being immutable).
+
+If the userTags field is changed post-install, there is no guarantee about how an in-cluster operator will respond to the change. Some operators may reconcile the change and change tags on the AWS resource. Some operators may ignore the change. If tags are removed from userTags, the tag may or may not be removed from the AWS resource.
+
+### User Stories
+
+https://issues.redhat.com/browse/SDE-1146 - IAM users and roles can only operate on resources with specific tags
+As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+### Upgrade / Downgrade Strategy
```

Comment:
On upgrade:
The new status field won't be populated since it is only populated by the installer and that can't have happened if the cluster was installed from a prior version.  Components that consume the new field should take no action since they will see no additional tags

On downgrade:
The status field may remain populated(i'm not actually sure what will get served from a CRD api perspective), components may or may not continue to tag newly created resources w/ the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.

Reply:
(the above is a suggested entry for the `upgrade/downgrade strategy` section)

Reply:
Agree with above  , less concerned during experimental.

Reply:
added, thanks!